scripts/geofence_email_notifications.py

```diff
@@ -0,0 +1,128 @@
+    # Create the GIS
+    logger.info("Authenticating...")
+    # First step is to get authenticate and get a valid token
+    gis = GIS(arguments.org_url,
+              username=arguments.username,
+              password=arguments.password,
+              verify_cert=not arguments.skip_ssl_verification)
+    
+    if arguments.lkl_layer_url:
+        lkl_layer = FeatureLayer(url=arguments.lkl_layer_url, gis=gis)
+    else:
+        logger.info("Please pass an LKL layer url!")
+        sys.exit(0)
+    
+    logger.info("Checking LKL values exist")
+    # Query LKL and check features exist
+    lkl_fset = lkl_layer.query('1=1')
+    if len(lkl_fset) == 0:
+        logger.info("No LKLs in your layer yet!")
+        sys.exit(0)
+    
+    logger.info("Buffering around the provided lat/long ")
+    pt = Point({"x": args.longitude, "y": args.latitude, "spatialReference": {"wkid": 4326}})
+    buffer = arcgis.geometry.buffer([pt], distances=args.distance_tolerance, in_sr=4326, out_sr=4326, unit=9001)
+    buffer_filter = arcgis.geometry.filters.intersects(buffer[0], sr=4326)
+    results = lkl_layer.query(geometry_filter=buffer_filter).features
+    if len(results) > 0:
+        # ensure yagmail is set up before you hit this part of the code: https://github.com/kootenpv/yagmail#username-and-password
+        yag = yagmail.SMTP(args.gmail_user)
+        recipient_emails = arguments.recipient_emails.replace(" ", "").split(",")
+        body = ""
+        users = []
+        for result in results:
+            users.append(result.attributes['created_user'])
+            body = body + f"As of {pendulum.from_timestamp(result.attributes['last_edited_date'] / 1000)}, the user {result.attributes['created_user']} is in the area that is {args.distance_tolerance} meters around ({args.latitude},{args.longitude})\n"
+        subject = str(users)[1: -1] + " in geofence"
+        yag.send(to=recipient_emails, subject=subject, contents=body)
+        logger.info("Email sent")
+    else:
+        logger.info("No users in the geofence")
+    logger.info("Completed!")
+
+
+if __name__ == "__main__":
+    # Get all of the commandline arguments
+    parser = argparse.ArgumentParser("Email if a Tracker user is in a geofence")
+    parser.add_argument('-u', dest='username', help="The username to authenticate with", required=True)
+    parser.add_argument('-p', dest='password', help="The password to authenticate with", required=True)
+    parser.add_argument('-org', dest='org_url', help="The url of the org/portal to use", required=True)
+    # Parameters for tracker
+    parser.add_argument('-lkl-layer-url', dest='lkl_layer_url', required=True,
+                        help="The last known location (LKL) layer (either location tracking service or tracks view) you'd like to use. This URL should end in /1")
+    parser.add_argument('-latitude', dest='latitude',type=float,required=True, help="Latitude of the point you want to buffer")
```

Comment:
I think ideally the input would be a feature layer of polygons. There might be a `description` field the user can specify that could be used in the message.